File: JavaExamples/src/example2/PrinterImpl.java
package example2;

public class PrinterImpl implements Printer {
	private final static String prefix = "[JAVA] ";
	public String sentence;

	public PrinterImpl(String sentence) {
		this.sentence = sentence;
	}

	@Override
	public void print() {
		System.out.println(prefix + this.sentence.trim());
	}
}

File: Interop/src/printer/PrinterImpl.kt
package printer

open class PrinterInteropImpl(val prefix: String?) : Printer {
	override fun print(sentence: String?) = if(sentence is String) safePrint(sentence) else printNull()
	
	fun safePrint(sentence: String) = println(this.prefix + sentence.trim())
	fun printNull() = println(this.prefix + "The sentence is NULL, but we are safe.")
}